src/components/Borrow/borrowUtils.tsx

```diff
@@ -0,0 +1,460 @@
+  assetFrom: AssetPriceData,
+  cRatio: string,
+  vaultEntryAmountCollateral: BN,
+  amountCollateral: BN,
+  vaultEntryAmountBorrow: BN,
+  minCRatio: string
+) => {
+  const amountInputAfterCalculation = calculateAmountBorrow(
+    assetTo.priceVal,
+    assetTo.assetScale,
+    assetFrom.priceVal,
+    assetFrom.assetScale,
+    amountCollateral,
+    cRatio
+  )
+  const amountVaultAfterCalculation = calculateAmountBorrow(
+    assetTo.priceVal,
+    assetTo.assetScale,
+    assetFrom.priceVal,
+    assetFrom.assetScale,
+    vaultEntryAmountCollateral,
+    minCRatio
+  )
+
+  if (amountVaultAfterCalculation.sub(vaultEntryAmountBorrow).gte(new BN(0))) {
+    return amountInputAfterCalculation.add(amountVaultAfterCalculation.sub(vaultEntryAmountBorrow))
+  } else {
+    return amountInputAfterCalculation
+  }
+}
+
+export const calculateAvailableWithdraw = (
+  assetTo: AssetPriceData,
+  assetFrom: AssetPriceData,
+  cRatio: string,
+  vaultEntryAmountCollateral: BN,
+  amountSynthetic: BN,
+  vaultEntryAmountBorrow: BN
+) => {
+  const amountAfterCalculation = calculateAmountCollateral(
+    assetTo,
+    assetFrom,
+    vaultEntryAmountBorrow.sub(amountSynthetic),
+    cRatio
+  )
+  if (
+    amountAfterCalculation.gte(new BN(0)) &&
+    vaultEntryAmountCollateral.sub(amountAfterCalculation).gte(new BN(0))
+  ) {
+    return vaultEntryAmountCollateral.sub(amountAfterCalculation)
+  } else {
+    return new BN(0)
+  }
+}
+
+export const calculateLiqAndCRatio = (
+  action: string,
+  priceFrom: BN,
+  amountCollateral: BN,
+  vaultAmountCollatera: BN,
+  priceTo: BN,
+  amountBorrow: BN,
+  vaultAmountBorrow: BN,
+  liqThreshold: Decimal,
+  assetScaleTo: number,
+  assetScaleFrom: number
+) => {
+  const ratioTo = calculateCRatio(
+    priceTo,
+    assetScaleTo,
+    priceFrom,
+    assetScaleFrom,
+    action === 'borrow' ? amountBorrow.add(vaultAmountBorrow) : vaultAmountBorrow.sub(amountBorrow),
+
+    action === 'borrow'
+      ? amountCollateral.add(vaultAmountCollatera)
+      : vaultAmountCollatera.sub(amountCollateral)
+  )
+
+  const ratioFrom = calculateCRatio(
+    priceTo,
+    assetScaleTo,
+    priceFrom,
+    assetScaleFrom,
+    vaultAmountBorrow,
+    vaultAmountCollatera
+  )
+  return {
+    liquidationTo: Number(
+      calculateLiqPrice(
+        priceFrom,
+        action === 'borrow'
+          ? amountCollateral.add(vaultAmountCollatera)
+          : vaultAmountCollatera.sub(amountCollateral),
+        priceTo,
+        action === 'borrow'
+          ? amountBorrow.add(vaultAmountBorrow)
+          : vaultAmountBorrow.sub(amountBorrow),
+        liqThreshold,
+        assetScaleTo,
+        assetScaleFrom
+      )
+    ),
+    liquidationFrom: Number(
+      calculateLiqPrice(
+        priceFrom,
+        vaultAmountCollatera,
+        priceTo,
+        vaultAmountBorrow,
+        liqThreshold,
+        assetScaleTo,
+        assetScaleFrom
+      )
+    ),
+    cRatioTo:
+      ratioTo === 'NaN'
```

Comment:
format differently these ternaries, they are not good to read, are mixing up, use some brackets